lib/redcord/connection_pool.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+require 'connection_pool'
+require_relative 'redis'
+
+class Redcord::ConnectionPool
+  def initialize(pool_size:, timeout:, **client_options)
+    @connection_pool = ::ConnectionPool.new(size: pool_size, timeout: timeout) do
+      # Construct a new client every time the block gets called
+      Redcord::Redis.new(**client_options, logger: Redcord::Logger.proxy)
+    end
+  end
+
+  # Avoid method_missing when possible for better performance
+  methods = Redcord::Redis.instance_methods(false) + Redis.instance_methods(false)
```

Comment:
does this contain duplication? Would `Redcord::Redis.instance_methods(false)` cover all the `Redis` instance_methods?

Reply:
According to https://ruby-doc.org/core-2.6.3/Module.html#method-i-instance_methods, if the optional parameter is false, the methods of any ancestors are not included. So `Redcord::Redis.instance_methods(false)` only covers the ones we define, unless we maybe overwrite a `Redis` method (though the correct one should still lbe called, since connection_pool is created with `Redcord::Redis` objects).

I can make this into a set anyway just in case.